homeassistant/components/suez_water/coordinator.py

```diff
@@ -48,61 +34,38 @@ def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         )
 
     async def _async_setup(self) -> None:
-        self._sync_client = await self.hass.async_add_executor_job(self._get_client)
+        self._async_client = await self._get_client()
+        self._data_api = SuezData(self._async_client)
 
     async def _async_update_data(self) -> AggregatedSensorData:
         """Fetch data from API endpoint."""
         async with asyncio.timeout(30):
-            return await self.hass.async_add_executor_job(self._fetch_data)
+            try:
+                data = await self._fetch_data()
+                _LOGGER.debug("Successfully fetched suez data")
+            finally:
+                await self._async_client.close_session()
```

Comment:
Why do we keep closing the session? Ideally we support passing in a client session and just keep it open